pkg/generator/provisioner.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generator
+
+// ProvisioningUtility specifies the type of utility used for machine provisioning
+type ProvisioningUtility string
+
+const (
+	Ignition  ProvisioningUtility = "ignition"
+	CloudInit ProvisioningUtility = "cloud-init"
+)
+
+// GetProvisioningUtility returns the provisioning utility for the given machine
+func GetProvisioningUtility(osFlavor string) ProvisioningUtility {
+	switch osFlavor {
+	case "flatcar":
```

Comment:
Would be nice to have the operating systems that we support as an enum, instead of plain text